***Solving a linear system Ax = b with (random) 7000 × 7000 matrix A***

In [21]:
set.seed(123)
n <- 7000

A <- matrix(runif(n * n), nrow = n, ncol = n)
b <- runif(n)

execution_time <- system.time(replicate(70, solve(A, b)))[["elapsed"]]
cat(paste("R Linear System Mean Time:", execution_time / 70, "seconds"))

R Linear System Solution Time: 17.1452857142857 seconds

first attempt: R Linear System Mean Time: 12.5982857142858 seconds

second attempt: R Linear System Mean Time: 17.1452857142857 seconds

***Computing the eigenvalues/vectors of (random) 1200 × 1200 matrix A***


In [23]:
set.seed(123)
n <- 1200

A <- matrix(runif(n * n), nrow = n, ncol = n)

compute_eigen <- function() {
  start_time <- Sys.time()
  eigen_result <- eigen(A)
  elapsed_time <- difftime(Sys.time(), start_time, units = "secs")
  return(elapsed_time)
}

execution_times <- replicate(70, compute_eigen())
mean_time <- mean(execution_times)
cat(paste("Eigenvalues and Eigenvectors Mean Time: ", mean_time, " seconds", sep = ""))

Eigenvalues and Eigenvectors Mean Time: 3.7308742421014 seconds

first attempt: Eigenvalues and Eigenvectors Mean Time: 3.7416442837034 seconds

second attempt: Eigenvalues and Eigenvectors Mean Time: 3.7308742421014 seconds

***Computing the singular values/vectors of a (random) 2100 × 2100
matrix A***

In [27]:
set.seed(42)

A <- matrix(runif(2100 * 2100), nrow = 2100, ncol = 2100)

num_iterations <- 70

total_execution_time <- 0

for (i in 1:num_iterations) {
  start_time <- Sys.time()
  
  svd_result <- svd(A)
  
  end_time <- Sys.time()
  
  execution_time <- end_time - start_time
  
  total_execution_time <- total_execution_time + execution_time
}

average_execution_time <- total_execution_time / num_iterations

singular_values <- svd_result$d

cat("Average execution time:", average_execution_time, "\n")

Average execution time: 9.742058 


first attempt: Average execution time: 12.78046 seconds

second attempt: Average execution time: 9.742058 seconds